### DISCUSSION TITLE
How to create a USB pyudev.Device with a given PID and VID for testing

### DISCUSSION THREAD

**Discussion by brianreinhold** [pyudev/pyudev#470] [Q&A]
I am using the MonitorObserver in a class to trigger some handler on a USB plugin event  with the correct VID and PID. I can call the callback method that the MonitorObserver calls but it takes as parameters 'action' and 'Device'.

I started this way in the test:
```
        device: pyudev.Device = pyudev.Device(pyudev.Context(), ?)
        watch.handle_usb_event('add', device)
```
I want to create a Device object such that
```
        vendor_id: str = device.get('ID_VENDOR_ID')
        device_id: str = device.get('ID_MODEL_ID')
```
returns VIDs and PIDs that I determine. Does anyone know how to do this? (I am using unittest as the framework.)

**Comment by mulkieran** [comment]
```device.get``` actually invokes the underlying C library code to get the property information from udev. If I were testing, I would try dynamically patching the ```Properties.__getitem__``` method.

**Reply by brianreinhold** [reply]
So just monkey patch Properties.__getitem__('some text') with my own get method that returns what I want? That way I get the 'real' Device object from pyudev and don't have to worry about creating a mock.

**Reply by mulkieran** [reply]
@brianreinhold I think that should work for you.
